commit d70da12453ac3797e0c54884305ccc894e8c817b
Author: Kees Cook <kees@kernel.org>
Date:   Fri Mar 7 20:29:26 2025 -0800

    hardening: Enable i386 FORTIFY_SOURCE on Clang 16+
    
    The i386 regparm bug exposed with FORTIFY_SOURCE with Clang was fixed
    in Clang 16[1].
    
    Link: https://github.com/llvm/llvm-project/commit/c167c0a4dcdb998affb2756ce76903a12f7d8ca5 [1]
    Reviewed-by: Nathan Chancellor <nathan@kernel.org>
    Link: https://lore.kernel.org/r/20250308042929.1753543-2-kees@kernel.org
    Signed-off-by: Kees Cook <kees@kernel.org>

diff --git a/arch/x86/Makefile b/arch/x86/Makefile
index 4c136bcaff86..3ba7e185924e 100644
--- a/arch/x86/Makefile
+++ b/arch/x86/Makefile
@@ -137,7 +137,7 @@ ifeq ($(CONFIG_X86_32),y)
         include $(srctree)/arch/x86/Makefile_32.cpu
         KBUILD_CFLAGS += $(cflags-y)
 
-    ifeq ($(CONFIG_CC_IS_CLANG),y)
+    ifneq ($(call clang-min-version, 160000),y)
         # https://github.com/llvm/llvm-project/issues/53645
         KBUILD_CFLAGS += -ffreestanding
     endif